llvm/lib/Analysis/VectorUtils.cpp

```diff
@@ -1477,6 +1479,31 @@ void VFABI::getVectorVariantNames(
   }
 }
 
+FunctionType *VFABI::createFunctionType(const VFInfo &Info,
+                                        const FunctionType *ScalarFTy) {
+  // Create vector parameter types
+  SmallVector<Type *, 8> VecTypes;
+  ElementCount VF = Info.Shape.VF;
+  for (auto [Idx, VFParam] : enumerate(Info.Shape.Parameters)) {
+    if (VFParam.ParamKind == VFParamKind::GlobalPredicate) {
+      VectorType *MaskTy =
+          VectorType::get(Type::getInt1Ty(ScalarFTy->getContext()), VF);
+      VecTypes.push_back(MaskTy);
+      continue;
+    }
+
+    Type *OperandTy = ScalarFTy->getParamType(Idx);
```

Comment:
As before I think this is only safe because the global predicate is last.  This is why my example maintained a separate parameter count for the scalar function parameters.